openstacknagios/neutron/Routers.py

```diff
@@ -0,0 +1,114 @@
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#  
+"""
+   Nagios plugin to check router status. Router status is an extended feature
+   provided to neutron via astara. https://github.com/openstack/astara . This
+   plugin will **NOT** work for neutron servers without astara extensions.
+
+   This corresponds to the output of 'neutron router-list -c id -c status'.
+"""
+
+import openstacknagios.openstacknagios as osnag
+
+import keystoneclient.v2_0.client as ks
+
+from neutronclient.neutron import client
+
+
+class NeutronRouters(osnag.Resource):
+    """
+    Determines the number down/build/active routers
+
+    """
+
+    def __init__(self, args=None):
+        self.openstack = self.get_openstack_vars(args=args)
+        osnag.Resource.__init__(self)
+
+    def probe(self):
+        try:
+            k = ks.Client(username=self.openstack['username'],
+                          password=self.openstack['password'],
+                          tenant_name=self.openstack['tenant_name'],
+                          auth_url=self.openstack['auth_url'],
+                          cacert=self.openstack['cacert'],
+                          insecure=self.openstack['insecure'])
+        except Exception as e:
+            self.exit_error('cannot get token ' + str(e))
+         
+        try:
+            neutron = client.Client('2.0',
+                                    endpoint_url=k.service_catalog.url_for(
+                                        endpoint_type='public',
+                                        service_type='network'),
+                                    token=k.auth_token, 
+                                    ca_cert=self.openstack['cacert'],
+                                    insecure=self.openstack['insecure'])
+        except Exception as e:
+            self.exit_error('cannot load ' + str(e))
+
+        try:
+            result = neutron.list_routers()
+        except Exception as e:
+            self.exit_error(str(e))
+
+        stati = dict(active=0, down=0, build=0)
+
+        for router in result['routers']:
+            if router['status'] == 'ACTIVE':
+                stati['active'] += 1
+            if router['status'] == 'DOWN':
+                stati['down'] += 1
+            if router['status'] == 'BUILD':
+                stati['build'] += 1
+
+        for r in stati.keys():
+            yield osnag.Metric(r, stati[r], min=0)
+
+
+@osnag.guarded
+def main():
+    argp = osnag.ArgumentParser(description=__doc__)
+
+    argp.add_argument('-w', '--warn', metavar='RANGE', default='0:',
+                      help="""return warning if number of down routers is
+                      greater than (default: 0)""")
+    argp.add_argument('-c', '--critical', metavar='RANGE', default='10:',
+                      help="""return critical if number of down routers is
+                      greater than (default: 10) """)
+    argp.add_argument('--warn_build', metavar='RANGE', default='0:',
```

Comment:
This default seems a little aggressive to me - it's pretty common for routers to go into build from time to time (like whenever somebody adds an IP).